homeassistant/components/amcrest/__init__.py

```diff
@@ -222,68 +102,126 @@ def _start_event_monitor(hass, name, api, event_codes):
     thread.start()
 
 
-def setup(hass, config):
+async def async_setup(hass, config):
+    """Old way to set up Amcrest devices."""
+
+    # Import from yaml if exist
+    conf = config.get(DOMAIN)
+    if conf is not None:
+        for entry in conf:
+            # Only import if we haven't before.
+            config_entry = _async_find_matching_config_entry(hass)
```

Comment:
You're only returning a single entry here. So why is this done inside the for loop ?